ViewModel/BrowserViewModel.swift

```diff
@@ -45,41 +44,74 @@ class BrowserViewModel: NSObject, ObservableObject,
     @Published private(set) var outlineItems = [OutlineItem]()
     @Published private(set) var outlineItemTree = [OutlineItem]()
     @Published private(set) var url: URL?
-    
+    @Published var externalURL: URL?
+
     let tabID: NSManagedObjectID?
     let webView: WKWebView
     private var canGoBackObserver: NSKeyValueObservation?
     private var canGoForwardObserver: NSKeyValueObservation?
     private var titleURLObserver: AnyCancellable?
     private var bookmarkFetchedResultsController: NSFetchedResultsController<Bookmark>?
-    
+    private let scriptHandler: BrowserScriptHandler
+    private let navDelegate: BrowserNavDelegate
+    private let uiDelegate: BrowserUIDelegate
+    /// A temporary placeholder for the url that should be opened in a new tab, set on macOS only
+    static var urlForNewTab: URL?
+    private var cancellables: Set<AnyCancellable> = []
+
     // MARK: - Lifecycle
-    
+
     init(tabID: NSManagedObjectID? = nil) {
         self.tabID = tabID
-        self.webView = WKWebView(frame: .zero, configuration: WebViewConfiguration())
+        webView = WKWebView(frame: .zero, configuration: WebViewConfiguration())
+        scriptHandler = BrowserScriptHandler()
+        navDelegate = BrowserNavDelegate()
+        uiDelegate = BrowserUIDelegate()
         super.init()
-        
+
+        scriptHandler.$outlineItems.assign(to: \.outlineItems, on: self)
+            .store(in: &cancellables)
+        scriptHandler.$outlineItemTree.assign(to: \.outlineItemTree, on: self)
+            .store(in: &cancellables)
+
+        navDelegate.$externalURL.assign(to: \.externalURL, on: self)
+            .store(in: &cancellables)
+
+        uiDelegate.$externalURL.assign(to: \.externalURL, on: self)
+            .store(in: &cancellables)
+
+        uiDelegate.$createBookMark.sink { [weak self] url in
+            self?.createBookmark(url: url)
+        }.store(in: &cancellables)
+
+        uiDelegate.$deleteBookMark.sink { [weak self] url in
+            self?.deleteBookmark(url: url)
+        }.store(in: &cancellables)
```

Comment:
I disagree with splitting one view model class into three. Because now you have the same amount of logic (yes I know it's a lot), but scattered in four places. So you might have more specialized classes, but it is harder to get to the actual logic (so arguably less readable)

And this block of code above (line 72-89), there are more code to maintain to achieve the same results, so sorry I cannot get behind that

Reply:
I guess this came because of the Codefactor error, maybe here we should then just add an exception.

Reply:
Reverted to one file with SwiftLint ignores.